### ISSUE TITLE
segmentation fault when use `radian` in neovim's embedded terminal 

### ISSUE THREAD

**Issue by milanglacier** [neovim/neovim#17903]
### Neovim version (nvim -v)

v0.7.0-dev+1294-g0c02e8a62

### Vim (not Nvim) behaves the same?

no 8.2.3489

### Operating system/version

macOS 12.2.1

### Terminal name/version

iterm2 3.4.15

### $TERM environment variable

xterm-256color

### Installation

homebrew build from source

### How to reproduce the issue

0. Have `radian` installed:[radian](https://github.com/randy3k/radian)

1. `nvim --clean` to start a minimal vanilla neovim.
2. `:set mouse=a` this is optinal to enable mouse support.
3. `:e xxx.txt` open any files.
4. `:sp` to vertical split the screen.
5. `<C-w> j` type control-w simulaneously and then release and type j, to move the cursor to the bottom split of the screen.
6. `:term` to open the terminal embedded in neovim.
7. `i` to enter the insert mode.
8. `radian` to enter the radian REPL.
9. `<C-\><C-n>` to exit the insert mode, and return to normal mode. 
10. `<C-w> 100-` to shorten the width of radian's window
11. `<C-w> 20 +` to widen the width of radian's window
12. repeat 9, 10. an error may be triggered: `99564 segmentation fault` and neovim crashes.

if step 2 is done, step 10-12 can be replaced by simple using mouse to drag the upper frame of the radian's window.

Basically the problem is: if the window of radian is too small, it has the possiblity to suddenly crashed the entire neovim.
I've tested with other REPLs, including vanilla R, ipython, julia, they are fine, and I can arbitrarily drag the window.

This is not only happens in iterm2, it also happens in vimR, the GUI. But what happens for vimR is it totally freezes, and not quit back to the shell prompt. And the stable version $0.6.1$ also has this issue.

### Expected behavior

I can arbitrarily drag the window, just like using other terminal tools.

### Actual behavior

https://user-images.githubusercontent.com/45728125/160515356-2c13936b-0c89-4397-b308-65fc314cc90c.mov

**Comment by zeertzjq** [comment]
Backtrace:
```
    #2 0x564d94c1b072 in erase **/.deps/build/src/libvterm/src/state.c:47:9
    #3 0x564d94c1d8c0 in on_csi **/.deps/build/src/libvterm/src/state.c:1023:9
    #4 0x564d94c17cd6 in do_csi **/.deps/build/src/libvterm/src/parser.c:36:9
    #5 0x564d94c18239 in vterm_input_write **/.deps/build/src/libvterm/src/parser.c:278:9
    #6 0x564d94887ca1 in terminal_receive **/src/nvim/terminal.c:705:3
    #7 0x564d936f286e in on_channel_output **/src/nvim/channel.c:625:7
    #8 0x564d936ef8fb in on_channel_data **/src/nvim/channel.c:604:3
    #9 0x564d93aa4752 in read_event **/src/nvim/event/rstream.c:192:5
    #10 0x564d93aa39cc in invoke_read_cb **/src/nvim/event/rstream.c:205:3
    #11 0x564d93aa25bc in read_cb **/src/nvim/event/rstream.c:134:3
    #12 0x564d94c0aece in uv__read **/.deps/build/src/libuv/src/unix/stream.c:1247:7
    #13 0x564d94c0b817 in uv__stream_io **/.deps/build/src/libuv/src/unix/stream.c:1315:5
    #14 0x564d94c13a72 in uv__io_poll **/.deps/build/src/libuv/src/unix/epoll.c:374:11
    #15 0x564d94c00473 in uv_run **/.deps/build/src/libuv/src/unix/core.c:391:5
    #16 0x564d93a8df6a in loop_poll_events **/src/nvim/event/loop.c:61:3
    #17 0x564d942f8a82 in inbuf_poll **/src/nvim/os/input.c:440:3
    #18 0x564d942f8514 in os_inchar **/src/nvim/os/input.c:115:19
    #19 0x564d947ce122 in state_enter **/src/nvim/state.c:62:13
    #20 0x564d9411c39a in normal_enter **/src/nvim/normal.c:463:3
    #21 0x564d93eaaf93 in main **/src/nvim/main.c:572:3
    #22 0x7f30fdbfd30f in __libc_start_call_main libc-start.c
    #23 0x7f30fdbfd3c0 in __libc_start_main@GLIBC_2.2.5 (/usr/lib/libc.so.6+0x2d3c0)
    #24 0x564d932f5f74 in _start (**/build/bin/nvim+0xbcff74)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV **/.deps/build/src/libvterm/src/screen.c:278:22 in erase_internal
==1866318==ABORTING
```

**Comment by seandewar** [comment]
Looks like #16699

**Comment by zeertzjq** [comment]
Replacing libvterm-0.1 with libvterm 0.2 indeed seems to fix the crash.

Duplicate of #16699

**Comment by milanglacier** [comment]
> Replacing libvterm-0.1 with libvterm 0.2 indeed seems to fix the crash.
> 
> Duplicate of #16699

Then how to do that ? How to specify 0.2 version of libvterm? I build from source using `homebrew`'s formula automatically......

**Comment by zeertzjq** [comment]
It is not always straightforward to do. That's why #16699 hasn't been closed yet.

**Comment by milanglacier** [comment]
> It is not always straightforward to do. That's why #16699 hasn't been closed yet.

OK. Seems like I still need wait for sometimes until everything is merged?

**Comment by zeertzjq** [comment]
The current PR is #17329. Unfortunately it is blocked by a libvterm bug.

**Comment by zeertzjq** [comment]
I just tested the reproducing steps on latest HEAD builtin with libvterm v0.3 and this crash no longer happens.
